velocity/src/main/java/host/bloom/ab/velocity/VelocityPlugin.java

```diff
@@ -54,8 +52,7 @@ public void onProxyInitialization(ProxyInitializeEvent event) {
         new VelocityCommandHandler(this, this.server.getCommandManager());
 
         // We can either use the built-in event or our own channel listener
-        if (this.config.catchRawConnections) {
-
+        if (ConfigManager.getConfig().getProperty(ConfigKeys.catch_raw_connections)) {
```

Comment:
We need to make sure `VelocityLoginListener` is initialized so `onJoin` and `onQuit` for the actionbar works 
I guess we can just pass whether we are catching raw connections and just return `onLoginInbound`

Reply:
The event registration can be moved in the try catch since it returns anyway

Reply:
I made some changes, VelocityLoginListener is always initialized however it'll only check for the ConnectionHandshakeEvent if catch raw connections is not enabled.

If I am understanding how this setup is suppose to work, that should be fine? The method in the hook channel "initChannel" doesn't seem to have any usages.

Reply:
Yes, that looks good, thanks! We just replace the channel initializer with our wrapper so we can get a more accurate connection count, but this looks good